### PR TITLE
React/error boundary

### PR DESCRIPTION
**Pull request by dpinol**
## Description

We want that if CustomMessage crashes, it will be replaced by the specified component (typically a simple error message). So far the solution only works if the failure occurs while deserializing the component. Pending to understand why it does not working yet if it fails during rendering of a new component.

## Context
CustomMessage can fail if there's a bug in their implementation. A typical case is when restoring a serialized component from the browser local storage.

## To document / Usage example

React [ErrorBoundary](https://reactjs.org/docs/error-boundaries.html)

## Testing

The pull request...

- has integration tests

### CODE REVIEW

File: packages/botonic-react/src/components/components.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+/**
+ * Get the name of a component TYPE
+ * @param {React.ComponentType} component
+ * @returns {string}
+ */
+export function getComponentTypeName(component) {
+  return component.displayName || component.name || 'Component'
+}
```
Comment (vanbasten17) [inline packages/botonic-react/src/components/components.js:17]: move to `util/components.js`?

**Review by ericmarcos — APPROVED** [review]
This is so cool! 👏👏👏

**Review by vanbasten17 — APPROVED** [review]
💯 💯 💯

File: packages/botonic-react/src/components/components.js (hunk 1 above)
Comment (dpinol) [reply packages/botonic-react/src/components/components.js:17]: will do soon

**Pull request merged**